src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/FileUtils.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.hellbender.tools.spark.sv.utils;
+
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+
+public final class FileUtils {
+
+    public static void writeSAMFile(final Iterator<SAMRecord> alignments, final SAMFileHeader header, final String outputName,
+                             final boolean preOrdered) {
+        try ( final SAMFileWriter writer = createSAMFileWriter(outputName, header, preOrdered) ) {
+            alignments.forEachRemaining(writer::addAlignment);
+        } catch ( final UncheckedIOException ie) {
+            throw new GATKException("Can't write SAM file to the specified location: " + outputName, ie);
+        }
+    }
+
+    private static SAMFileWriter createSAMFileWriter(final String samFile, final SAMFileHeader header, final boolean preOrdered) {
```

Comment:
What about `ReadUtils.createCommonSAMWriter()` -- Could that be used in place of this? This seems like something that should be handled by some engine code.